### DISCUSSION TITLE
js-api map.graphics.fromGeoJson

### DISCUSSION THREAD

**Discussion by marko-jer** [e-netze/webgis-community#200] [General]
Hallo,
ich habe ein Karte aus dem Mapbuilder `"Erstellen: html_dokument"` erstellt und mit der JS-API [https://docs.webgiscloud.com/dev/API/API_JavaScript/graphics.htmll#fromgeojsonanchor](url) angereichert. Die Koordinaten für eine Kreis werden als Parameter übergeben.
Leider funktioniert es seit mindestens 5.24.1801 nicht mehr auch nicht mit 6.xx.xxx .
Ich habe eine alte API-Solo Instanz 5.22.3001, mit der es noch funktioniert.

Hier zwei Beispiellinks:
[https://gis.stmk.gv.at/content/dokumente/sites/GraphicsFromGeoJson5.22.3001.html?x=533191&y=5211512](url)
[https://gis.stmk.gv.at/content/dokumente/sites/GraphicsFromGeoJson5.24.1801.html?x=533191&y=5211512](url)

Hat sich da was geändert? 

lg

**Comment by gis-eni** [comment]
Das Problem scheinr aufzutreten, weil die Karte noch nicht komplett fertig geladen ist, wenn du dein geoJSON hinzufügst.
Ein schneller Workaround ist, wenn du eine Logik erst verzögert startest:

``` javascript

webgis.init(function() {
    var map = webgis.createMap('map', {
        ...
        });
  
    map.ui.createHourGlass(); 
   // map.ui Dinge

   webgis.delayed(function() {
      // deine Logik
      if(xExist) {
          // code
      } else {
         // code
      }
   },500);  // 500 ms verzögert (sollte immer reichen)
```

Für die Zukunft kann man für dieses Problem einmal ein ``onInitialized`` Event für das ``map`` Objekt einführen. Dazu sind aber noch ein paar Überlegungen notwendig
